src/frontend/src/components/AddUser/index.tsx

```diff
@@ -12,26 +12,38 @@ interface Props {
 interface State {
   email: string;
   password: string;
+  confirmPassword: string;
   name: string;
   group: string;
+  search: string;
+  admin: string;
   role: string;
   invalidCredentials: boolean;
   invalidResponse: boolean;
+  missingName: boolean;
   invalidEmail: boolean;
   missingPassword: boolean;
+  invalidPassword: boolean;
+  mismatchPasswords: boolean;
 }
 
 class AddUser extends React.Component<Props, State> {
   public state: State = {
     email: '',
     password: '',
+    confirmPassword: '',
     name: '',
     group: '',
+    search: '',
```

Comment:
`search` and `admin` are possible values for the `role` variable, ie: `role = 'search'` or `role = 'admin'` so they can be removed from state.